### PR TITLE
Feature/sm 19 generate mapping from pipeline

### PR DESCRIPTION
**Pull request by ievgenvarava**
Added writer stage to save pipeline contents into file

### CODE REVIEW

File: src/SprykerMiddleware/Zed/Process/Business/ProcessFacadeInterface.php [hunk 1]
```diff
@@ -13,4 +13,28 @@
...
+     */
+    public function translate(array $payload, array $dictionary);
+
+    /**
+     * @param array $payload
+     * @param string $destination
+     *
+     * @return array
+     */
+    public function writeSerialized(array $payload, string $destination);
```
Comment (alex-galych) [inline src/SprykerMiddleware/Zed/Process/Business/ProcessFacadeInterface.php]: I think the process shouldn't know about concrete details, like writers type. Should be like writePayload and 
 a writer has to be selected based on a configuration.

File: src/SprykerMiddleware/Zed/Process/Business/ProcessFacade.php [hunk 2]
```diff
@@ -35,4 +35,30 @@ public function translate(array $payload, array $dictionary)
...
             ->translate($payload);
     }
+
+    /**
+     * @param array $payload
+     * @param string $destination
+     *
+     * @return array
+     */
+    public function writeSerialized(array $payload, string $destination)
```
Comment (ievgenvarava) [inline src/SprykerMiddleware/Zed/Process/Business/ProcessFacade.php]: i would be better to have single write() method and determine format with parameters

**Review by ievgenvarava — COMMENTED** [review]
remove format-dependent methods from ProcessFacade

**Review by valerio8787 — APPROVED** [review]

File: src/SprykerMiddleware/Zed/Process/Business/Writer/JsonFileWriter.php [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+{
+    /**
+     * @param array $payload
+     * @param string $destination
+     *
+     * @return array
+     */
+    public function write(array $payload, string $destination): array
+    {
+        file_put_contents($destination, json_encode($payload));
```
Comment (lazystar) [inline src/SprykerMiddleware/Zed/Process/Business/Writer/JsonFileWriter.php]: we might want to go through the filesystem module in a later iteration.

File: src/SprykerMiddleware/Zed/Process/Business/Exception/WriterNotFoundException.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+     *
+     * @return string
+     */
+    protected function buildMessage(string $className)
+    {
+        $message = 'Spryker Middleware Exception' . PHP_EOL;
+        $message .= sprintf(
+            'Can not resolve %1$s writer',
+            $className
+        ) . PHP_EOL;
```
Comment (lazystar) [inline src/SprykerMiddleware/Zed/Process/Business/Exception/WriterNotFoundException.php]: why this strange mixture of concatenation and `sprintf`?

File: src/SprykerMiddleware/Zed/Process/Business/Writer/JsonFile.php [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+{
+    /**
+     * @param array $payload
+     * @param string $destination
+     *
+     * @return array
+     */
+    public function write(array $payload, string $destination): array
+    {
+        file_put_contents($destination, json_encode($payload));
```
Comment (lazystar) [inline src/SprykerMiddleware/Zed/Process/Business/Writer/JsonFile.php:15]: We might want to go through the filesystem module in a later iteration.

**Review by lazystar — APPROVED** [review]

**Pull request merged**